sdk/notificationhubs/notification-hubs/src/auth/sasTokenProvider.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  AccessToken,
+  NamedKeyCredential,
+  SASCredential,
+  isNamedKeyCredential,
+  isSASCredential,
+} from "@azure/core-auth";
+import { isObjectWithProperties } from "@azure/core-util";
+import { signString } from "./hmacSha256.js";
+
+/**
+ * A SasTokenProvider provides an alternative to TokenCredential for providing an `AccessToken`.
+ * @hidden
+ */
+export interface SasTokenProvider {
+  /**
+   * Property used to distinguish SasTokenProvider from TokenCredential.
+   */
+  isSasTokenProvider: true;
```

Comment:
nit: maybe a `kind: 'sasTokenProvider'` would be more extensible in the future?

Reply:
Just following the pattern from other token providers.  Perhaps we need to revisit them all?